utils.py

```diff
@@ -103,6 +103,104 @@ def store_boe_pdfs(base_url, minutes_url):
     return
 
 
+def get_boe_minutes(base_url="https://comptroller.baltimorecity.gov", minutes_url_part="/boe/meetings/minutes", save_folder='BoE Minutes PDFs', start_dir=os.path.dirname(__file__)):
+    """
+    base_url:
+        url for the comptrollers website
+
+    minutes_url:
+        the extra bit of the URL that specifies the minutes "landing page"
+
+        https://comptroller.baltimorecity.gov/boe/meetings/minutes
+
+    save_folder:
+        the name of the folder used to hold the minutes PDFs, shouldn't have any slashes
+
+    start_dir:
+        defaults to the directory the calling script is in. used to make a full path (relative to the calling script or absolute) with save_folder
+    """
+    # print(start_dir)  # print the directory this script is in
+    os.chdir(start_dir)  # set the cwd to avoid any OS/python shenanigans
+    pdf_save_dir = f'{start_dir}/{save_folder}/'  # set save directory for pdfs
+    os.makedirs(pdf_save_dir, exist_ok=True)  # create directory if it doesn't exist (no error if it does)
+    minutes_url = base_url + minutes_url_part
+
+    # because whoever wrote the BoE website didn't make it easy to scrape
+    # date_regex = re.compile(r'([(January|February|March|April|May|June|July|August|September|October|November|December)]) (\d{1,2}), (\d{4})', re.IGNORECASE)
+    date_regex = re.compile(r'(\w*)\s+(\d{1,2})\D*(\d{4})', re.IGNORECASE)
+    month_dict = {'january': '01',
+                  'february': '02',
+                  'march': '03',
+                  'april': '04',
+                  'may': '05',
+                  'june': '06',
+                  'july': '07',
+                  'august': '08',
+                  'september': '09',
+                  'october': '10',
+                  'november': '11',
+                  'december': '12'
+                }
+
+    # this creates a dictionary containing all possible single-letter deletions of every month
+    # it's used to correct for typos in the text of a link to the minutes
+    # ex: Novembe 17, 2010 (one of two real examples as of Aug 9, 2020)
+    typo_dict = {k: list() for k in month_dict.keys()}
+    for month in typo_dict.keys():
+        for i in range(len(month)):
+            typo_dict[month].append(month[:i] + month[i+1:])
+
+    start_page = requests.get(minutes_url)  # grab the 'starting page' that provides links to the minutes for a specific year
+    start_page.raise_for_status()  # doing the \
+    start_soup = BS(start_page.text, 'lxml')  # needful
+
+    # this eliminates the need to specify the years to grab since four-digit years are used consistently
+    year_tags = start_soup.find_all('a', href=True, text=re.compile(r'^\d{4}$'))  # find the tags that link to the minutes for specific years
```

Comment:
We could change the regex to r'^20\d{2}$' and then the link text would have to be a valid year in the 21st century. Beyond that, we could add some checking when downloading the minutes to make sure any minutes links are downloading PDFs. I'm not sure this is going to be an issue even if another four-digit year link (that's not to a minutes page) shows up because it would attempt to scrape that site but wouldn't find anything (or only a few things). Either way, if it _does_ come up, it shouldn't be a complicated fix.

Reply:
I think the correct strategy here would actually be to localize the search to the specific div in which the years appear, the same could be said for the retrieval of the minute links. Currently we rely heavily on the format of the href or the anchor tag itself rather than leveraging the structure of the page itself. Based on what we've seen so far, the link formats are likely to change because they seem to be entered manually, by contrast the structure of the page seems auto-generated. 

The solution that is balances simplicity with robustness to capture the largest proportion of minutes will likely have to employ both strategies (i.e. page sections and link formatting)